pkg/apis/maistra/v2/smcp_new.yaml

```diff
@@ -296,20 +279,29 @@ spec:
       some-other-egress-gateway: {}
 
   runtime:
+    components:
     # component specific overrides
-    pilot: # same as gateway.runtime structure illustrated above
-      deployment:
-        replicas: 2
-      pod:
-        affinity: {}
-        containers:
-          discovery:
-            resources:
-              limits: {}
-              requirements: {}
-    citadel: {} # similar to pilot above
-    galley: {} # similar to pilot above
-    # note, mixer runtime is configured as part of policy.mixer or telemetry.mixer
+      pilot: # same as gateway.runtime structure illustrated above
+        deployment:
+          replicas: 2
```

Comment:
this is just an example file embedded with the struct definition, as it's difficult to get an idea of the structure by looking at the go code (since it's all spread around between files).

also, i think we've decided that we want the profiles to be _the_ default, so we're not relying on values.yaml.  that said, i don't think we'll be there for a while.  also, it means the user will have to have a full config if they're not using the default profile, which means we may want to always apply a default.

Reply:
the default profile should be used as the base for *everything* shouldn't it? If you don't specify a profile, use default. If you specify a profile, have it inherit at some level from default? I don't think that should be avoidable.

Reply:
Thanks.